src/web/app/pages/ConnectMachineIp/index.tsx

```diff
@@ -23,31 +25,32 @@ import { adorModels, promarkModels } from 'app/actions/beambox/constant';
 import { allWorkareas } from 'app/constants/workarea-constants';
 import { IDeviceInfo } from 'interfaces/IDevice';
 
-import styles from './ConnectMachineIp.module.scss';
+import { swiftrayClient } from 'helpers/api/swiftray-client';
+import styles from './index.module.scss';
+import { initialState, State } from './state';
+import Hint from './Hint';
+
+const SWIFTRAY_RECONNECT_INTERVAL = 5000;
+const MACHINE_CONNECTION_TIMEOUT = 30;
 
 // TODO: add test
 const ConnectMachineIp = (): JSX.Element => {
   const lang = useI18n();
   const [ipValue, setIpValue] = useState('');
-  const [state, setState] = useState<{
-    testState: TestState;
-    countDownDisplay: number;
-    device: IDeviceInfo | null;
-  }>({
-    testState: TestState.NONE,
-    countDownDisplay: 0,
-    device: null,
-  });
+  const [state, setState] = useState<State>(initialState);
+  const [swiftrayReadyState, setSwiftrayReadyState] = useState(swiftrayClient.readyState);
```

Comment:
In my opinion, the `swiftrayReadyState` does not need to be a state since it is not used in the rendering process.
Can use `const readyState = swiftrayClient.readyState;` directly in `checkSwiftrayConnection`